cpp/libcudf_kafka/include/cudf_kafka/kafka_consumer.hpp

```diff
@@ -149,9 +149,11 @@ class kafka_consumer : public cudf::io::datasource {
   /**
    * @brief Query the Kafka broker for the assigned topic Metadata
    *
+   * @param[in] Retrieves metadata for only a specific topic.
+   *
    * @return Map of Kafka topics with their corresponding topic partition values.
    */
-  std::map<std::string, std::vector<int32_t>> list_topics() const;
+  std::map<std::string, std::vector<int32_t>> list_topics(std::string specific_topic) const;
```

Comment:
If you are only getting a single topic, why does this need to return a map?

Reply:
If this only gets metadata for a single topic, why is it called "list topics"?

Reply:
Since custreamz extends Python Streamz we are bound by previously defined function names to ensure our library works with theirs. This just happens to be one of those functions names that maps to what they have and we can't change.

Reply:
I guess on the C++ side we could change the name and leave the python name as is but that almost seemed more confusing. I can do that if desired however.

Reply:
In libcudf we always try to do the right thing rather than doing the wrong thing because somebody else does it.  On the C++ side we have the freedom to do the right thing.

In the long term I think it's better to build a clear and usable API. If we don't, we end up spending way more refactoring later.

Reply:
> If this only gets metadata for a single topic, why is it called "list topics"?

If no topic name is specified, it gets the metadata for all topics. But we need to provide functionality to retrieve metadata for specific topics as well, since the metadata for all topics might not be needed at all times and can be a pretty large object at times.

The name `list_topics` is the name of the standard Confluent Kafka Python library function, which `custreamz.kafka` or `cudf_kafka` serves to be a drop-in replacement for in order to use GPUs instead of CPUs.

Reply:
This is not what the function documentation says. Why not make the function signature like this:

```
std::map<std::string, std::vector<int32_t>> list_topics(std::vector<std::string> const& topics) const;
```

Then the user can query a subset of topics (not just one or all). Document that if `topics` is empty, metadata for all topics is returned.  This API will be no more difficult to use, since the user can just call `list_topics({topic_name})` to convert a single name to a vector.

Reply:
I will name it whatever the group wants to name it. That part really doesn't matter to me. @harrism if this was coming from scratch I love your idea about a `std::vector` however the way the underlying C API is written right now only a single topic_name at a time is accepted https://docs.confluent.io/2.0.0/clients/librdkafka/classRdKafka_1_1Handle.html#ac0cc63c574ce027a184640645151a75c Now that isn't a big deal as I could of course iterate over the std::vector of topics and call that function multiple times but that isn't how folks using the API currently do it.

In a nut shell doesn't matter to me just wanted to lay out all the facts. Let me know which you prefer and I will take that approach.

Reply:
OK, I think leaving the API is OK, but please improve the documentation to be clear and accurate. Thanks!